src/Tools/dotnet-stack/Symbolicate.cs

```diff
@@ -0,0 +1,392 @@
+            public string Assembly;
+            public string Token;
+            public string Offset;
+            public string Document;
+            public string Filepath;
+            public string Filename;
+            public string StartLine;
+            public string EndLine;
+        }
+
+        private static void GetLineFromStack(IConsole console, List<string> xmlList, string inputPath, string outputPath)
+        {
+            try
+            {
+                List<string> outputString = new List<string>();
+                using StreamReader fsr = new StreamReader(new FileStream(inputPath, FileMode.Open, FileAccess.Read));
+                while (!fsr.EndOfStream)
+                {
+                    string line = fsr.ReadLine();
+                    if (!line.Contains("at ") || !line.Contains("+"))
+                    {
+                        outputString.Add(line);
+                        console.Out.WriteLine($"{line}");
+                        continue;
+                    }
+                    string ret = GetRegex(line, xmlList);
+                    outputString.Add(ret);
+                    console.Out.WriteLine($"{ret}");
+                }
+
+                string output = string.Empty;
+                if (outputPath != null)
+                {
+                    using StreamWriter fsw = new StreamWriter(new FileStream(outputPath, FileMode.Create, FileAccess.Write));
+                    foreach (var str in outputString)
+                    {
+                        fsw.WriteLine(str);
+                    }
+                    output = $"\nOutput: {outputPath}\n";
+                }
+                console.Out.WriteLine($"{output}");
+            }
+            catch (Exception e)
+            {
+                Console.Error.WriteLine(e);
+            }
+        }
+
+        private static string GetRegex(string line, List<string> xmlList)
+        {
+            string ret = line;
+            string logtagStr = Regex.Match(line, "(?<splitdata>.*?) at").Groups["splitdata"].Value;
+            if (logtagStr.Length != 0)
+            {
+                ret = line.Replace(logtagStr, "");
+            }
+            string typeMethodStr = Regex.Match(line, "at (?<splitdata>.*?)\\((.*)\\)").Groups["splitdata"].Value;
+            string methodStr = typeMethodStr.Split(".")[typeMethodStr.Split(".").Length - 1];
+            string typenameStr = typeMethodStr.Replace("." + methodStr, "");
+            string parameterStr = Regex.Match(line, methodStr + "\\((?<splitdata>.*?)\\)").Groups["splitdata"].Value;
+            string assemblyStr = Regex.Match(line, " in (?<splitdata>.*?)\\: ").Groups["splitdata"].Value;
+            string[] tokenOffsetStr = Regex.Match(line, "\\: token (?<splitdata>.*)?").Groups["splitdata"].Value.Split("+");
+            string xmlStr = assemblyStr.Contains(".ni.dll") ? assemblyStr.Replace(".ni.dll", ".xml") : assemblyStr.Replace(".dll", ".xml");
```

Comment:
I think you can do all this with one Regex that you can precompile. That should save on allocations, string copies, and other stuff.

Not sure if this gets everything, but this is a good starting point:
```csharp
var rx = new Regex(@"at (?<type>[\w+\.?]+)\.(?<method>\w+)\((?<params>.*)\) in (?<filename>[\w+\.?]+)(\.dll|\.ni\.dll): token (?<token>0x\d+)\+(?<offset>0x\d+)", RegexOptions.Compiled);
```